pkg/tsdb/opentsdb/opentsdb.go

```diff
@@ -3,53 +3,122 @@ package opentsdb
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
 	"path"
 	"strconv"
 	"strings"
 	"time"
 
-	"golang.org/x/net/context/ctxhttp"
-
 	"encoding/json"
-	"io/ioutil"
-	"net/http"
-	"net/url"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/infra/httpclient"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
+	"github.com/grafana/grafana/pkg/registry"
 	"github.com/grafana/grafana/pkg/setting"
+	"golang.org/x/net/context/ctxhttp"
 )
 
 type OpenTsdbExecutor struct {
 	httpClientProvider httpclient.Provider
+
+	im  instancemgmt.InstanceManager
+	cfg *setting.Cfg
+}
+
+type OpenTsdbService struct {
+	HTTPClientProvider   httpclient.Provider   `inject:""`
+	Cfg                  *setting.Cfg          `inject:""`
+	BackendPluginManager backendplugin.Manager `inject:""`
+}
+
+type datasourceInfo struct {
+	HTTPClient        *http.Client
+	Url               string `json:"url"`
+	BasicAuth         bool   `json:"basicAuth"`
+	BasicAuthUser     string `json:"basicAuthUser"`
+	BasicAuthPassword string `json:"basicAuthPassword"`
```

Comment:
I think we should skip this information and not setting the basic auth on outgoing request here, rather let the HTTP client provider handle that. 

We add a Basic Auth middleware to the HTTP client provider ([code](https://github.com/grafana/grafana/blob/503c3b8f66343f4bc49e0985081c810764f18b66/pkg/infra/httpclient/httpclientprovider/http_client_provider.go#L25)) that will set basic auth to outgoing request if  Authorization HTTP header is not already set ([code](https://github.com/grafana/grafana-plugin-sdk-go/blob/ff48557e2bcee9250906d1c5f863cf6be62c8cc3/backend/httpclient/basic_auth_middleware.go#L21))

Reply:
I see now, should I just create a new httpclient.Provider with the middleware, using the New function form the code, to set the the basic Auth middleware to the HTTP client provider?